core/cli/serialization.go

```diff
@@ -0,0 +1,78 @@
+package cli
+
+import (
+	"encoding/json"
+	"io"
+	"time"
+
+	"go.polydawn.net/repeatr/api/def"
+
+	"github.com/inconshreveable/log15"
+	"github.com/ugorji/go/codec"
+)
+
+var _ io.Writer = &journalSerializer{}
+
+type ctxPairs []interface{}
+
+func (ctxPairs) MapBySlice() {}
+
+type logItem struct {
+	Level int       `json:"level"`
+	Msg   string    `json:"message"`
+	Ctx   ctxPairs  `json:"ctx"`
+	Time  time.Time `json:"time"`
+}
+
+type runRecordOutput struct {
+	Data  interface{} `json:"runRecord"`
+	RunID def.RunID   `json:"runID"`
+}
+type journalOutput struct {
+	Data  string    `json:"journal"`
+	RunID def.RunID `json:"runID"`
+}
+type logOutput struct {
+	Data logItem `json:"log"`
+}
```

Comment:
```go
type row struct {
    RunRecord RunRecord `json:"runRecord,omitEmpty"`
    Journal string `json:"journal,omitEmpty"
    // etc...
}
```

may be an easier way to do this: implicitly a union struct; it wastes some memory, but makes deserialization a lot easier by kind of sidestepping polymorphism.